tap/tests/test_result.py

```diff
@@ -70,3 +80,24 @@ def test_adds_unexpected_success(self):
         self.assertEqual(
             line.directive.text, "TODO {}".format(_("(unexpected success)"))
         )
+
+    def test_adds_subtest_success(self):
+        """Test that the runner handles subtest success results."""
+        result = self._make_one()
+        test = FakeTestCase()
+        with test.subTest():
+            result.addSubTest(test, test._subtest, None)
+        line = result.tracker._test_cases["FakeTestCase"][0]
+        self.assertTrue(line.ok)
+
+    def test_adds_subtest_failure(self):
+        """Test that the runner handles subtest failure results."""
+        result = self._make_one()
+        # Python 3 does some extra testing in unittest on exceptions so fake
+        # the cause as if it were raised.
+        ex = Exception()
+        ex.__cause__ = None
+        test = FakeTestCase()
+        with test.subTest():
+            result.addSubTest(test, test._subtest, (ex.__class__, ex, None))
+        self.assertEqual(len(result.tracker._test_cases["FakeTestCase"]), 1)
```

Comment:
Do you mind switching this assert to check that the test line is a "not ok" line? I think that would be a more accurate assert for the spirit of this test.
```suggestion
        line = result.tracker._test_cases["FakeTestCase"][0]
        self.assertFalse(line.ok)
```